diesel_compile_tests/tests/compile-fail/right_side_of_left_join_requires_nullable.rs

```diff
@@ -27,13 +27,17 @@ fn main() {
     let join = users::table.left_outer_join(posts::table);
 
     // Invalid, only Nullable<title> is selectable
-    let _ = join.select(posts::title); //~ ERROR E0277
+    let _ = join.select(posts::title);
```

Comment:
```
error[E0271]: type mismatch resolving `<users::table as diesel::query_source::AppearsInFromClause<posts::table>>::Count == diesel::query_source::Succ<diesel::query_source::Never>`
  --> src/main.rs:30:18
   |
30 |     let _ = join.select(posts::title);
   |                  ^^^^^^ expected struct `diesel::query_source::Never`, found struct `diesel::query_source::Succ`
   |
   = note: expected type `diesel::query_source::Never`
              found type `diesel::query_source::Succ<diesel::query_source::Never>`
```

This is really hard to understand what's going on here. Is there any way this could be better ?